### PR TITLE
[WIP]fix the error when add comment between conditions

### PR DESCRIPTION
**Pull request by rchaser53**
fix: #3678

### CODE REVIEW

File: src/pairs.rs [hunk 1]
```diff
@@ -5,8 +5,9 @@ use crate::config::IndentStyle;
 use crate::rewrite::{Rewrite, RewriteContext};
 use crate::shape::Shape;
 use crate::utils::{
-    first_line_width, is_single_line, last_line_width, trimmed_last_line_width, wrap_str,
+    first_line_width, is_single_line, last_line_width, mk_sp, trimmed_last_line_width, wrap_str,
 };
+use syntax::source_map::BytePos;
```
Comment (topecongiro) [inline src/pairs.rs]: Please move this import statement under `use syntax::ast;`.

File: src/pairs.rs [hunk 2]
```diff
@@ -89,6 +93,23 @@ fn rewrite_pairs_one_line<T: Rewrite>(
...
 }
 
+fn set_pair_string<'a>(
+    result: &'a mut String,
+    comment: &Option<String>,
+    base_shape: Shape,
+    context: &RewriteContext<'_>,
+) -> &'a String {
+    if let Some(comment) = comment {
+        let indent_str = base_shape.indent.to_string_with_newline(context.config);
```
Comment (topecongiro) [inline src/pairs.rs]: Could you try to keep the original position of the comment, please? E.g.,

```rust
    long_expression_0 // post comment 0
        // pre comment 1
        * long_expression_1 // post commnt 1
        // pre comment 2
        * long_expression_2 // post commnt 2
        // pre comment 3
        * long_expression_3 // post commnt 3
        // pre comment 4
        * long_expression_4 // post commnt 4
        // pre comment 5
        * long_expression_5 // post commnt 5
        // pre comment 6
        * long_expression_6 // post commnt 6
```

given the above expression, each pre and post comment should stay as is.

**Comment by topecongiro** [comment]
@rchaser53 Pinging for triage - do you have any update?

**Comment by rchaser53** [comment]
It takes the time more for me. I close this PR and will send again.

**Pull request closed without merging**